applications/plugins/Geomagic/src/Geomagic/initPlugin.cpp

```diff
@@ -33,60 +35,62 @@
 #define PLUGIN_DATA_DIR_ QUOTE(PLUGIN_DATA_DIR)
 #endif
 
-namespace sofa
-{
 
-namespace component
+namespace geomagic
 {
+    extern void registerGeomagicDriver(sofa::core::ObjectFactory* factory);
+    extern void registerGeomagicEmulator(sofa::core::ObjectFactory* factory);
 
-	//Here are just several convenient functions to help user to know what contains the plugin
+    //Here are just several convenient functions to help user to know what contains the plugin
 
-	extern "C" {
+    extern "C" {
                 SOFA_GEOMAGIC_API void initExternalModule();
                 SOFA_GEOMAGIC_API const char* getModuleName();
                 SOFA_GEOMAGIC_API const char* getModuleVersion();
                 SOFA_GEOMAGIC_API const char* getModuleLicense();
                 SOFA_GEOMAGIC_API const char* getModuleDescription();
-                SOFA_GEOMAGIC_API const char* getModuleComponentList();
-	}
+                
+                SOFA_GEOMAGIC_API void registerObjects(sofa::core::ObjectFactory* factory);
+    }
 	
 
     void initExternalModule()
     {
         static bool first = true;
         if (first) {
             first = false;
+            // make sure that this plugin is registered into the PluginManager
+            sofa::helper::system::PluginManager::getInstance().registerPlugin(MODULE_NAME);
```

Comment:
@fredroy @alxbilger , I saw in several plugins like BeamAdapter that initExternalModule is calling a member method:
```void SOFA_BEAMADAPTER_API initBeamAdapter();```

which has a SOFA_EXPORT_DYNAMIC_LIBRARY macro, is this mandatory? Is it to ba able to register objects from a third party library?